### PR TITLE
[knx] Fix DPT 3.007 (Control Dimming) for non -control channels

### LINKED ISSUE #6970: [knx] Dimmer Type has UNDEF state after relativ dimming to 100%
Hello,

when I dim (relativ) my lights with a Gira Tastsendor Plus to 100% and release the button, the openhab dimmer item get the state "UNDEF".
It's no problem with dimming to 10, 33, 68 and so on percent. But if the state reachs 100%, I release the finger from the button and the item changes from 100 to UNDEF.

After reaching 100% and releasing the button I expected that the item state keeps 100%.

I've attached an ETS trace of the relevant GA's and some screenshots from ETS with further information of the GA.
Also an OH2 event.log of the corresponding dim event (from ETS).

Here are some ENV details:

```
# dpkg -l|grep openhab
ii  openhab2                             2.5.0-1                      all          openhab2
ii  openhab2-addons                      2.5.0-1                      all          openhab2-addons
# lsb_release -a
No LSB modules are available.
Distributor ID: Debian
Description:    Debian GNU/Linux 10 (buster)
Release:        10
Codename:       buster
# java -version
java version "1.8.0_161"
Java(TM) SE Runtime Environment (build 1.8.0_161-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.161-b12, mixed mode)
```

My OH2 items, things ...

```
knx.things
Type dimmer : R05_Deckenleuchte-Essen_Dimmen "R.05 Deckenleuchte-Essen Dimmen" [ switch="1/2/17+1/2/18", position="1/2/20+<1/2/21", increaseDecrease="1/2/19" ]

knx_licht.items
Dimmer R05_Dimmer_Esszimmer "Esszimmer [%s]" (gR05, gLicht) { channel="knx:device:router1:generic:R05_Deckenleuchte-Essen_Dimmen" }

default.sitemap
Default item=R05_Dimmer_Esszimmer
```

Any help is appreciated.

BR
Mitch

### PR DESCRIPTION
**Pull request by holgerfriedrich**
Fixes #6970.
Carryover from smarthomej/addons#162.

Also-by: Jan N. Klug github@klug.nrw

@J-N-K Is the commit message / credit ok for you? I added you to the sign-off using an Also-by statement.

Refs: https://github.com/openhab/openhab-addons/issues/14502

### CODE REVIEW

File: bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java [hunk 1]
```diff
@@ -899,6 +899,7 @@ private String quantityTypeToDPTValue(QuantityType<?> qt, int mainNumber, int su
                         case 8:
                             return translator3BitControlled.getControlBit() ? UpDownType.DOWN : UpDownType.UP;
                     }
+                    break;
```
Comment (lolodomo) [inline bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java:902]: I see that a break is missing for cases 1 and 2. Please confirm that it is expected.

**Comment by lolodomo** [comment]
The conflict was certainly introduced by the other PR I just merged. You added a "break" for case 14 in that PR and the other PR removed all the case 14.

File: bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java (hunk 1 above)
Comment (holgerfriedrich) [reply bundles/org.openhab.binding.knx/src/main/java/org/openhab/binding/knx/internal/dpt/KNXCoreTypeMapper.java:900]: case 1 has default case with return statement, case 2 also ends with a return statement -> LGTM

**Comment by holgerfriedrich** [comment]
> The conflict was certainly introduced by the other PR I just merged. You added a "break" for case 14 in that PR and the other PR removed all the case 14.

@lolodomo rebased, should now be fine again, thanks

**Review by J-N-K — APPROVED** [review]

**Comment by holgerfriedrich** [comment]
@lolodomo please do not remove the references/credit when merging to main

**Review by lolodomo — APPROVED** [review]
LGTM, thank you

**Comment by lolodomo** [comment]
> @lolodomo please do not remove the references/credit when merging to main

I have only that when I ask the merge:
image

**Comment by holgerfriedrich** [comment]
@lolodomo thanks for providing the image. I will check later today to find out why github cuts off the messages....

**Comment by lolodomo** [comment]
I added manually the "also-by" to the merge comment.

**Comment by holgerfriedrich** [comment]
@lolodomo If I try a PR on my repository, the squash merge looks correct:
image

The resulting commit is also fine.
https://github.com/holgerfriedrich/openhab-addons/commits/test-upstream

Can you please check the settings for squash merge on the addons repository?
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/configuring-commit-squashing-for-pull-requests
I have no permission to do it. My repo is set to "default", and the PR contains only one commit. So I would expect the full commit message.

**Comment by lolodomo** [comment]
It is setup to "default to pull request title".

**Comment by lolodomo** [comment]
In webui repo, it is set to "default to pull request title and description. 
@J-N-K should be able to tell us how it is setup in the core repo.

**Comment by J-N-K** [comment]
In core it is set to "Default"

**Comment by lolodomo** [comment]
I just changed to "default message". Let see if this is better.

**Comment by jlaur** [comment]
I have also noticed recently that the suggested commit message is no longer composed of the individual commits.

**Comment by jlaur** [comment]
@openhab/add-ons-maintainers - if anyone changed the configuration recently, please see above comments.

**Comment by kaikreuzer** [comment]
@jlaur That might have been me - sorry for playing around there and thanks for changing it back! 👍

**Pull request merged**